onnxscript/rewriter/pattern.py

```diff
@@ -169,28 +170,43 @@ class OpPatternBuilder:
 
     def __init__(
         self,
-        opset_pattern: Pattern[str],
-        op_name_pattern: Pattern[str],
+        opset_pattern: OpsetPatternBuilder,
+        op_name: str | Pattern[str],
     ) -> None:
         self.opset_pattern = opset_pattern
-        self.op_name_pattern = op_name_pattern
+        self.op_name = op_name
 
     def __call__(self, *args, **kwargs):
-        # TODO(rama): Unify with convention used elsewhere.
         if "_num_outputs" in kwargs:
-            num_outputs = kwargs["_num_outputs"]
-            del kwargs["_num_outputs"]
+            raise ValueError("Use 'outputs' instead of '_num_outputs'.")
+        if "version" in kwargs:
+            raise ValueError(
+                "The pattern builder does not support 'version' keyword argument."
+            )
+        if "domain" in kwargs:
```

Comment:
Should we define the expected kwargs on top of this function or class as a note/comment? It seems like there are multiple kwargs could be passed in.

Reply:
I think we can capture it explicitly. Commented above

Reply:
Agree with the suggestion to capture it explicitly. I think it is also important to describe it in the documentation of the rewriter. The users will be those who write rewrite-rules, who may not be looking at this class, which is a builder class.

A related question (this is the same convention/trick used in _tape too): these names can clash with attribute-names used in ONNX spec. I wonder if we should use names like `_domain` which are less likely to clash with ONNX attribute-names?

Reply:
I personally think that _domain is better. _ also warns users that they are modifying something important. But I don't have strong preference.

Reply:
Just ideating: can we make op a callable that creates a partially applied function? E.g.

```
op(domain="com.microsoft").Gelu(...)
```

but I see we also have `outputs`, which doesn't fit the semantics of parameterizing op